router/transformer/transformer.go

```diff
@@ -103,8 +103,13 @@ func Init() {
 
 // Transform transforms router jobs to destination jobs
 func (trans *handle) Transform(transformType string, transformMessage *types.TransformMessageT) []types.DestinationJobT {
+	var destinationJobs types.DestinationJobs
+	transformMessageCopy := *transformMessage
```

Comment:
Is there a reason doing the shallow copy here? 🤔

During hydration we manipulate non root level fields, thus I don't see the benefit.

Reply:
Just don't want to mutate input parameters of this method. As a principle it is safer not to mutate input parameters, because you never know if and how the owner (caller) can use them later on. If a method is mutating input parameters, this introduces strong coupling & should be explicitly mentioned in the method's doc. But even so, a caller might not read the doc thus it can cause issues/bugs.

Reply:
As a sidenote, the struct that we are dehydrating is not the same as the struct we are hydrating

Reply:
**_Update:_** you are correct, a shallow copy isn't enough, thus adapted the Dehydrate method to return a proper copy of the `TransformMessageT`